### DISCUSSION TITLE
Suggested way to bulk update nodes of a tree

### DISCUSSION THREAD

**Discussion by cseickel** [MunifTanjim/nui.nvim#71] [Q&A] [answered]
First of all thanks for the awesome library! 

I am building a file tree and I am wondering what's the expected way to replace all the nodes in a tree with a new collection? Should I set `tree.nodes` to the new table of TreeNodes and call `tree:render()`, or should I just create a whole new tree and let it overwrite the buffer?

I am thinking about things like refreshing after a file/folder was changed, or just changing the root to a new path.

**Comment by MunifTanjim** [comment, marked as answer]
Hey, thanks for trying it out! 😃

Yeah, that's a use-case I forgot about when implementing it.

> Should I set `tree.nodes` to the new table of TreeNodes and call `tree:render()`

I think you shouldn't do that and only use the public APIs documented in the documentation.

> should I just create a whole new tree and let it overwrite the buffer?

~You can do that for now.~

~But the real solution would be to add an API like `tree:set_nodes(nodes)` that takes a list and replaces all the nodes. That would be useful for this use-case, right?~

> what's the expected way to replace all the nodes in a tree with a new collection? 

I just added the [`tree:set_nodes` method](https://github.com/MunifTanjim/nui.nvim/tree/fc09d2fa/lua/nui/tree#treeset_nodes), and that would be the recommended way of doing it.

**Comment by cseickel** [comment]
That's perfect, thanks!
